scene/resources/importer_mesh.cpp

```diff
@@ -1364,7 +1367,7 @@ void ImporterMesh::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_surface_name", "surface_idx", "name"), &ImporterMesh::set_surface_name);
 	ClassDB::bind_method(D_METHOD("set_surface_material", "surface_idx", "material"), &ImporterMesh::set_surface_material);
 
-	ClassDB::bind_method(D_METHOD("generate_lods", "normal_merge_angle", "normal_split_angle", "bone_transform_array"), &ImporterMesh::generate_lods);
+	ClassDB::bind_method(D_METHOD("generate_lods", "normal_merge_angle", "normal_split_angle", "bone_transform_array"), &ImporterMesh::_generate_lods_bind);
```

Comment:
If backwards compatibility wasn't a problem would you still use this approach?

We have a system for providing compatibility versions of methods (for GDextensions and C#) to maintain backwards compatibility. 

So if you think that exposing normal_weight to scripting makes sense, then we should do it and just provide the compatibility method as necessary

Reply:
I feel like it's not a question as to whether normal_weight should be exposed, but how it should be exposed. If the goal is to maintain compatibility and simultaneously allow future extensions - imagine 5 more floating point parameters here added over the next 5 years - I'd probably favor an interface where an object with the set of named parameters is passed, whether it's a dictionary or a custom type. If we were to extend `generate_lods` as it is with normal weight, we'd need to add it to the end of the list for GDScript compatibility (C# supports function overloading but GDScript doesn't from what I can tell); this is certainly possible, but this ordering feels unnatural due to the existing bone array argument. This is why I thought it would be better to, for now, limit this change to just the manual importer usage; ideally, another function (with a different name I'd guess...) could be added later that accepts a string->float dictionary or something along these lines?

I'm open to adding this as an optional fourth argument, it's just that the process of wrestling with the binding layer made me question whether that's a good idea so keeping the interface as is for now seemed better.

Reply:
There's been a repeated design where the engine exposes a dictionary of options.

Reply:
Is there a reason it can't be added to the end of the argument list with a default value? Is the whole compat issue because it was added to the middle of the argument list?

If we think it is complex or likely to change enough to require named parameters, in my opinion, the right way to do this is with a RefCounted object which can contain input (and extra output data) and can be reused between invocations (zero allocations)

Reply:
I'm open for either route here; it seemed reasonable to exclude this addition from the script binding to begin with , as then the extra parameter can be exposed to scripts separately in a way that people feel is most future proof, but if the consensus is that it's better to just add the extra argument to the end with a C# specific second fixed length overload, I can certainly do that.